gpjax/types.py

```diff
@@ -31,6 +34,46 @@ def in_dim(self) -> int:
     def out_dim(self) -> int:
         return self.y.shape[1]
 
+    def _make_tfd_if_none(self) -> None:
+        if self._tf_dataset is None:
+            self._tf_dataset = tfd.Dataset.from_tensor_slices((self.X, self.y))
+    
+    def cache(self, *args, **kwargs) -> "Dataset":
+        self._make_tfd_if_none()
+        self._tf_dataset = self._tf_dataset.cache(*args, **kwargs)
+        return self
+    
+    def repeat(self, *args, **kwargs) -> "Dataset":
+        self._make_tfd_if_none()
+        self._tf_dataset = self._tf_dataset.repeat(*args, **kwargs)
+        return self
+    
+    def shuffle(self, *args, **kwargs) -> "Dataset":
+        self._make_tfd_if_none()
+        self._tf_dataset = self._tf_dataset.shuffle( *args, **kwargs)
+        return self
+    
+    def batch(self, *args, **kwargs) -> "Dataset":
+        self._make_tfd_if_none()
+        self._tf_dataset = self._tf_dataset.batch(*args, **kwargs)
+        return self
+    
+    def prefetch(self, *args, **kwargs) -> "Dataset":
+        self._make_tfd_if_none()
+        self._tf_dataset = self._tf_dataset.prefetch(*args, **kwargs)
+        return self
+    
+    def get_batches(self) -> tp.Callable[[None], "Dataset"]:
```

Comment:
Should this be called something like `make_batch_fn()`, `get_batch_fn()`, or `get_batcher()`. The current `get_batches()` name makes me think I'm going to receive a set of batches.

Reply:
Absolutely -> renamed to 'get_batcher' in [254643d](https://github.com/thomaspinder/GPJax/pull/48/commits/254643d4c32cb8a899d4395bd1b2055dfc5648e4).